yt_dlp/extractor/bilibili.py

```diff
@@ -404,7 +415,7 @@ def _real_extract(self, url):
 
 
 class BiliBiliBangumiIE(BilibiliBaseIE):
-    _VALID_URL = r'https?://(?:www\.)?bilibili\.com/bangumi/play/(?P<id>ep\d+)/?(?:$|[#?])'
```

Comment:
Why delete `/?(?:$|[#?])` ?

Reply:
I tested `/?(?:$|[#?])`. It worked fine and solved [this problem.](https://github.com/yt-dlp/yt-dlp/pull/7337#issuecomment-1608850016)

Reply:
> Why delete `/?(?:$|[#?])` ?

Because it was unnecessary

> it ... solved [this problem](https://github.com/yt-dlp/yt-dlp/pull/7337#issuecomment-1608850016)

No, both URLs are matched without the `/?(?:$|[#?])`